lib/data/data_sources/remote_data_source/chat_data_source.dart

```diff
@@ -48,18 +52,19 @@ class ChatDataSource with ChannelEventHandler {
     }
   }
 
-  Future<BaseMessage> sendMessage(String message) async {
+  Future<BaseMessage?> sendMessage(String message) async {
+    BaseMessage userMessage;
     try {
-      final Completer<BaseMessage> completer = Completer();
       GroupChannel channel = await getGroupChannel(_channelUrl);
-      channel.sendUserMessageWithText(message, onCompleted: (baseMessage, err) {
+      userMessage = channel.sendUserMessageWithText(message,
+          onCompleted: (baseMessage, err) {
         if (err != null) {
```

Comment:
this seems to be wrong, if there is an error we are sending the message and if it's not we are sending the message too